src/main/java/graphql/execution/reactive/CompletionStageOrderedSubscriber.java

```diff
@@ -0,0 +1,84 @@
+package graphql.execution.reactive;
+
+import graphql.Internal;
+import org.reactivestreams.Subscriber;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+/**
+ * This subscriber can be used to map between a {@link org.reactivestreams.Publisher} of U
+ * elements and map them into {@link CompletionStage} of D promises, and it keeps them in the order
+ * the Publisher provided them.
+ *
+ * @param <U> published upstream elements
+ * @param <D> mapped downstream values
+ */
+@Internal
+public class CompletionStageOrderedSubscriber<U, D> extends CompletionStageSubscriber<U, D> implements Subscriber<U> {
```

Comment:
It inherents from CompletionStageSubscriber because again they are very much the same

Reply:
except for ordering of source events